src/main/kotlin/de/bund/digitalservice/useid/metrics/MetricsService.kt

```diff
@@ -0,0 +1,30 @@
+package de.bund.digitalservice.useid.metrics
+
+import de.bund.digitalservice.useid.tenant.PARAM_NAME_TENANT_ID
+import io.micrometer.core.instrument.Counter
+import io.micrometer.core.instrument.Metrics
+import org.springframework.stereotype.Service
+
+@Service
+class MetricsService {
+
+    fun incrementSuccessCounter(method: String, tenantId: String) {
+        createCounter(method, "200", tenantId).increment()
+    }
+
+    fun incrementErrorCounter(method: String, tenantId: String) {
+        createCounter(method, "500", tenantId).increment()
+    }
+
+    private fun createCounter(method: String, status: String, tenantId: String): Counter {
```

Comment:
Here we could check a list of existing counters with the key method,tenant_id and reuse that instead of creating a new one. Unless the Metrics.counter method already does that.

Reply:
Good idea. I have added this as a comment for improvement, since we want to merge first and then do more improvements.